superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,234 @@
+        return True
+
+    def _benchmark(self):
+        """Implementation for benchmarking."""
+        M = self._args.m
+        K = self._args.k
+        N = self._args.n
+        P = self._args.p
+        Q = self._args.q
+        kernel = self._args.kernel
+        if self.__local_rank == 0:
+            logger.info(
+                "Computation Communication Overlap - using {} GPUs, matrix shape for computation: M={} K={} N={}, message tensor shape of nccl = [{},{}], ratio between computation kernel and nccl kernel={}"
+                .format(self.__world_size, M, K, N, P, Q, self._args.ratio)
+            )
+
+        MatA = list()
+        MatB = list()
+        # Matrix A
+        for _ in range(self._args.ratio):
+            MatA.append(torch.randn(M, K).cuda())
+        # Matrix B
+        MatB = torch.randn(K, N).cuda()
+        # message for nccl to transport
+        shape = [P, Q]
+        message = torch.randn(*shape).cuda()
+
+        # warm up
+        for i in range(self._args.num_warmup):
+            if not self.__kernel_nccl_pipeline(kernel, MatA, MatB, self._args.ratio, message, times=100):
+                return False
+        torch.cuda.synchronize()
+
+        # run and collect results
+        start = time.perf_counter()
+        for i in range(self._args.num_steps):
+            self.__kernel_nccl_pipeline(kernel, MatA, MatB, self._args.ratio, message, times=100)
+        compute_end = time.perf_counter()
+        torch.cuda.synchronize()
+        end = time.perf_counter()
+
+        logger.info(
+            "GPU {} total compute cost {}".format(
+                self.__local_rank, (compute_end - start) * 1000 / self._args.num_steps
+            )
+        )
+
+        compute_metric = 'computation_communication_overlap_gpu{}_computation_cost_{}'.format(self.__local_rank, kernel)
+        total_metric = 'computation_communication_overlap_gpu{}_total_cost_{}'.format(self.__local_rank, kernel)
+        compute_elapse_times = [(compute_end - start) * 1000 / self._args.num_steps]
+        elapse_times = [(end - start) * 1000 / self._args.num_steps]
+        if not self._process_numeric_result(compute_metric, compute_elapse_times
+                                            ) or not self._process_numeric_result(total_metric, elapse_times):
+            return False
+
+        logger.info(
+            " GPU: {} {} pipeline mean cost of each stage with NCCL all_reduce, time: {} ms".format(
+                self.__local_rank, kernel, (end - start) * 1000 / self._args.num_steps
+            )
+        )
+        torch.distributed.destroy_process_group()
+        return True
+
+
+BenchmarkRegistry.register_benchmark(
+    'pytorch-computation-communication-overlap-matmul', ComputationCommunicationOverlap, parameters='--kernel matmul'
+)
+BenchmarkRegistry.register_benchmark(
+    'pytorch-computation-communication-overlap-mul', ComputationCommunicationOverlap, parameters='--kernel mul'
+)
```

Comment:
@guoshzhao can we merge them into one benchmark here?

Reply:
Yes, I think it's better to merge them into one benchmark. Just like allredue and allgather in ShardingMatmul. Just register it as:
BenchmarkRegistry.register_benchmark(
    'pytorch-computation-communication-overlap', ComputationCommunicationOverlap, parameters='--kernel matmul mul'
)

Reply:
my concern is should we always use the same shape for all kernels?

Reply:
my concern is should we always use the same shape for all kernels? @guoshzhao

Reply:
If we want to use another shape, we can add a same benchmark with different settings in user configs. It means your benchmark can be run two times with different shapes.